### PR TITLE
Fix fetch parameters not being applied correctly

### PR DESCRIPTION
**Pull request by xconverge**
Fixes https://github.com/nodejs/undici/issues/1864

### CODE REVIEW

File: lib/fetch/index.js [hunk 1]
```diff
@@ -1951,8 +1951,6 @@ async function httpNetworkFetch (
         body: fetchParams.controller.dispatcher.isMockActive ? request.body && request.body.source : body,
         headers: request.headersList[kHeadersCaseInsensitive],
         maxRedirections: 0,
-        bodyTimeout: 300_000,
-        headersTimeout: 300_000,
```
Comment (xconverge) [inline lib/fetch/index.js:1955]: this would overwrite what was set in the agent a few lines up in ```const agent = fetchParams.controller.dispatcher```

**Comment by xconverge** [comment]
Not sure how this directly undoes https://github.com/nodejs/undici/pull/1386 and the test still passes, will look into it and see if I am missing something obvious?

**Comment by mcollina** [comment]
@ronag @KhafraDev why is `fetch()` completely overriding the agent options? I think it'd be best to stick to what is configured in the agent.

(I think I missed https://github.com/nodejs/undici/pull/1386, otherwise I would have objected then. It was a busy period of my life.).

**Review by ronag — CHANGES_REQUESTED** [review]
I think the fix here is simply that fetch doesn't pass it's own timeout values. Why do we override them in fetch?

**Comment by KhafraDev** [comment]
They are overridden in fetch to match chrome's behavior. I agree that the fix is to remove them.

**Comment by xconverge** [comment]
Do I need to adjust any defaults anywhere to preserve behavior or simply delete the 2 lines in lib/fetch/index.js? And remove my 300 sec test that was aiming to enforce the previous feature request?

**Comment by KhafraDev** [comment]
removing the 2 lines is fine, along with the other changes

**Comment by xconverge** [comment]
Ok this is ready for a re-review, it was my original changeset as well until I realized it was undo-ing #1373 and #1386

If you want me to add the 300 second tests back in from my previous commit and change the defaults elsewhere in the agent, please let me know, but for now at least the ability to override in fetch exists/works and a test that makes sure of that is in (unless the default gets changed to a much larger number in the future)

**Comment by xconverge** [comment]
Oops sorry I didn't know clicking re-request review on a change request removes the other reviewers

**Review by KhafraDev — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by xconverge** [comment]
@mcollina @ronag what happens if you set bodyTimeout and headersTimeout to 20 minutes, but do NOT set keepAliveTimeout (default is 10 minutes), and a request takes say 15 minutes.

Looking at this code, it kind of looks to me like the request will just die and eventually timeout at the 20 minutes and return a bodyTimeout or headersTimeout. This is what I am experiencing (I know this is wack to have such long requests... but I am starting by just porting this request from axios which works)

```
    if (this.shouldKeepAlive && client[kPipelining]) {
      const keepAliveTimeout = this.keepAlive ? util.parseKeepAliveTimeout(this.keepAlive) : null

      if (keepAliveTimeout != null) {
        const timeout = Math.min(
          keepAliveTimeout - client[kKeepAliveTimeoutThreshold],
          client[kKeepAliveMaxTimeout]
        )
        if (timeout <= 0) {
          socket[kReset] = true
        } else {
          client[kKeepAliveTimeoutValue] = timeout
        }
      } else {
        client[kKeepAliveTimeoutValue] = client[kKeepAliveDefaultTimeout]
      }
    } else {
      // Stop more requests from being dispatched.
      socket[kReset] = true
    }
```

I will increase setting keepAliveTimeout but am wondering if there is a more elegant way to handle this so that the errors I get make sense

**Comment by mcollina** [comment]
I'm failing to understand how that relates to this PR. If not, please open a new issue.

**Comment by xconverge** [comment]
> I'm failing to understand how that relates to this PR. If not, please open a new issue.

Ok fair enough, I have little to no information to put in a new issue yet so was hoping to just get some guidance. I was hopeful that this PR fixed my long request issue and it does not, so to me there is something about undici that I continue to not understand for a long request. I am glad I was able to write a tangible test for this particular issue, so I will see if I can figure out how to write a test case for what I am still experiencing.

**Comment by xconverge** [comment]
This is why I thought it was related, because it allowed me to expose the next issue basically https://github.com/nodejs/undici/issues/1887

**Pull request merged**